cytotable/convert.py

```diff
@@ -245,34 +253,20 @@ def _get_table_chunk_offsets(
 
             return None
 
-    # find chunk offsets from sql statement
     elif sql_stmt is not None:
-        # gather the total rowcount from csv or sqlite data input sources
         with _duckdb_reader() as ddb_reader:
-            rowcount = int(
-                ddb_reader.execute(
-                    # nosec
-                    f"SELECT COUNT(*) FROM ({sql_stmt})"
-                ).fetchone()[0]
-            )
+            sql_query = f"SELECT {page_key} FROM ({sql_stmt}) ORDER BY {page_key}"
+            keys = ddb_reader.execute(sql_query).fetchall()
+            keys = [key[0] for key in keys]
 
-    return list(
-        range(
-            0,
-            # gather rowcount from table and use as maximum for range
-            rowcount,
-            # step through using chunk size
-            chunk_size,
-        )
-    )
+    return _generate_pagesets(keys, chunk_size)
 
 
 @python_app
 def _source_chunk_to_parquet(
```

Comment:
Should this function be renamed to not use chunk since based on what I can see, you don't use chunk_size anymore? Would `_source_pageset_to_parquet` be more accurate?

Reply:
Thanks, great point - I like that change and will make a further revision to avoid the use of "chunk" to defer to "page" where possible.

Reply:
Thinking about this more, I'm going to make minimal changes to the usage of those words. Generally I feel we need to distinguish between Parquet pages vs CytoTable pages, and retaining "chunk" might be more understandable for now (in addition to avoiding a change to the `convert` API.